src/compiler/builder.ts

```diff
@@ -867,14 +883,20 @@ function handleDtsMayChangeOfReferencingExportOfAffectedFile(
     }
 
     const seenFileAndExportsOfFile = new Set<string>();
+    // If exported const enum, we need to ensure that js files are emitted as well since the const enum value changed
+    const invalidateJsFiles = !!affectedFile.symbol?.exports && !!forEachEntry(
+        affectedFile.symbol.exports,
+        exported => (skipAlias(exported, state.program!.getTypeChecker()).flags & SymbolFlags.ConstEnum) !== 0,

```

Comment:
I see you’re skipping the alias to handle

```ts
declare const enum AWorker {
  ONE = 1
}
export { AWorker as A };
```

but this would also return true for

```ts
export { AWorker as A } from "./some-other-module";
```

Did you intend to return true even if the const enum declaration is in another file?